repos/base-linux/src/core/io_mem_session_component.cc

```diff
@@ -12,14 +12,60 @@
  */
 
 #include <base/log.h>
+#include <linux_dataspace/client.h>
+
+#include <core_linux_syscalls.h>
 
 #include <io_mem_session_component.h>
 
 using namespace Genode;
 
+size_t Io_mem_session_component::get_arg_size(const char *args)
+{
+    size_t size = Arg_string::find_arg(args, "size").ulong_value(0);
+    addr_t base = Arg_string::find_arg(args, "base").ulong_value(0);
+
+    addr_t end = align_addr(base + size, PAGE_SIZE_LN);
+    base = base & ~(PAGE_SIZE - 1);
+    return (size_t)(end - base);
+}
+
+addr_t Io_mem_session_component::get_arg_phys(const char *args)
+{
+    return Arg_string::find_arg(args, "base").ulong_value(0);
+}
+
+Cache_attribute Io_mem_session_component::get_arg_wc(const char *args)
+{
+    Arg a = Arg_string::find_arg("wc", args);
+    if(a.valid() && a.bool_value(0)){
+        return WRITE_COMBINED;
+    }else{
+        return UNCACHED;
+    }
+}
+
+Io_mem_session_component::Io_mem_session_component(Range_allocator *io_mem_alloc,
+                                                   Range_allocator *ram_alloc,
+                                                   Rpc_entrypoint  *ds_ep,
+                                                   const char      *args) :
+    _io_mem_alloc(io_mem_alloc),
+    _ds(get_arg_size(args), 0, get_arg_phys(args), get_arg_wc(args), true, 0),
+    _ds_ep(ds_ep)
+{
+    int _fd = -1;
+
+    _fd = lx_open("/dev/hwio", O_RDWR | O_SYNC);
```

Comment:
rename `_fd` to `fd` because it is not a private member but a local variable. Assign the return value of `lx_open` directly to `fd`. Turn `int fd` into `int const fd`.